Chatkit/Redux/State/RoomListState.swift

```diff
@@ -0,0 +1,14 @@
+
+struct RoomListState: State {
```

Comment:
Is there a reason I'm missing why this `RoomListState` is necessary rather than working with just a `[RoomState]` directly?

Reply:
That's required, because reducers are not fit for mapping between wire model and internal state. In redux this should be done using middleware, but we don't have such layer.

Reply:
We could potentially work around this issue, but this would probably involve introducing a concept of sub-actions. What are your thoughts on this problem?

Reply:
I don't think I fully understand the points you're making, very sorry 🥺.  Perhaps I need to read up on middleware.

I had a go and was able to get rid of the `RoomListState` entity without much effort.  See here...
https://github.com/pusher/chatkit-swift/commit/8e1ab00594e36ab9274151d1bf99de850bb46888

Perhaps you are suggesting that we will require the `RoomListState` so that **in future** we have the option of using "middleware"?

Reply:
I agree that this is possible to code, but the change you suggested breaks the definition of a reducer (and perhaps that's ok, because we don't want to follow redux?). 

Reducer should take an action and a state and return a new/unmodified state. In order to get rid of `RoomListState`, which annoys me as well, we would need to pass to the reducer a list of states (like you suggested) or break the action into sub-actions. Either of these approaches feels like a deviation from the classical redux. Are we ok with that?

Reply:
I not sure how its break the definition?  `[RoomState]` is itself a `State` as defined here...

https://github.com/pusher/chatkit-swift/commit/8e1ab00594e36ab9274151d1bf99de850bb46888#diff-a2e538b46ffc792fe7c83fd97e1d1284R22

Reply:
`extension Array: State where Element == RoomState {}` is a less implicit version of `RoomListState`.

I have replaced `RoomListState` with the above extension.